h2o-core/src/main/java/water/Job.java

```diff
@@ -364,7 +364,16 @@ private static class AssertNoKey extends MRTask<AssertNoKey> {
     @Override public void reduce( AssertNoKey ank ) { _found |= ank._found; }
   }
 
-  public static class JobCancelledException extends RuntimeException {}
+  public static class JobCancelledException extends RuntimeException {
+
+    public JobCancelledException() {
+      this(null);
+    }
+    
+    public JobCancelledException(Job job) {
```

Comment:
had several leakages on training timeout, so added this constructor to provide better error report.